administrator/components/com_guidedtours/src/Model/StepModel.php

```diff
@@ -0,0 +1,391 @@
+            $table->default  = $value;
+            $table->store();
+        }
+
+        // Clean the cache
+        $this->cleanCache();
+
+        return true;
+    }
+
+    /**
+     * Method to test whether a record can have its state changed.
+     *
+     * @param   object  $record  A record object.
+     *
+     * @return  boolean  True if allowed to change the state of the record.
+     * Defaults to the permission set in the component.
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function canEditState($record)
+    {
+        $user = Factory::getUser();
+        $app = Factory::getApplication();
+        $context = $this->option . '.' . $this->name;
+        $extension = $app->getUserStateFromRequest($context . '.filter.extension', 'extension', null, 'cmd');
+
+        if (!\property_exists($record, 'tour_id')) {
+            $tourID          = $app->getUserStateFromRequest($context . '.filter.tour_id', 'tour_id', 0, 'int');
+            $record->tour_id = $tourID;
+        }
+
+        // Check for existing tour.
+        if (!empty($record->id)) {
+            return $user->authorise('core.edit.state', $extension . '.state.' . (int) $record->id);
+        }
+
+        // Default to component settings if tour isn't known.
+        return $user->authorise('core.edit.state', $extension);
+    }
+
+    /**
+     * Method to get a table object, load it if necessary.
+     *
+     * @param   string $name    The table name. Optional.
+     * @param   string $prefix  The class prefix. Optional.
+     * @param   array  $options Configuration array for model. Optional.
+     *
+     * @return Table  A Table object
+     *
+     * @since  __DEPLOY_VERSION__
+     * @throws \Exception
+     */
+    public function getTable($name = '', $prefix = '', $options = [])
+    {
+        $name = 'step';
+        $prefix = 'Table';
+
+        if ($table = $this->_createTable($name, $prefix, $options)) {
+            return $table;
+        }
+
+        throw new \Exception(Text::sprintf('JLIB_APPLICATION_ERROR_TABLE_NAME_NOT_SUPPORTED', $name), 0);
+    }
+
+    /**
+     * Method to change the published state of one or more records.
+     *
+     * @param   array    &$pks   A list of the primary keys to change.
+     * @param   integer  $value  The value of the published state.
+     *
+     * @return  boolean  True on success.
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function publish(&$pks, $value = 1)
+    {
+        $table = $this->getTable();
+        $pks   = (array) $pks;
+        $app = Factory::getApplication();
+        $extension = $app->getUserStateFromRequest('com_guidedtours.state.filter.extension', 'extension', null, 'cmd');
+
+        // Default item existence checks.
+        if ($value != 1) {
+            foreach ($pks as $i => $pk) {
+                if ($table->load($pk) && $table->default) {
+                    $app->enqueueMessage(Text::_('COM_WORKFLOW_MSG_DISABLE_DEFAULT'), 'error');
```

Comment:
wrong error message - this component doesnt use workflows

Reply:
Removed the function